src/Entity/Transaction.php

```diff
@@ -0,0 +1,170 @@
+    public const PAYBOX_CONNECTION_FAILED = '00001';
+    public const PAYBOX_INTERNAL_ERROR = '00003';
+    public const PAYBOX_CARD_NUMBER_INVALID = '00004';
+    public const PAYBOX_CARD_END_DATE_INVALID = '00008';
+    public const PAYBOX_CARD_UNAUTHORIZED = '000021';
+    public const PAYBOX_PAYMENT_PAGE_TIMEOUT = '000030';
+
+    /**
+     * @var int
+     *
+     * @ORM\Column(type="integer", options={"unsigned"=true})
+     * @ORM\Id
+     * @ORM\GeneratedValue(strategy="AUTO")
+     */
+    private $id;
+
+    /**
+     * @var string|null
+     *
+     * @ORM\Column(length=100, nullable=true)
+     */
+    private $payboxResultCode;
+
+    /**
+     * @var string|null
+     *
+     * @ORM\Column(length=100, nullable=true)
+     */
+    private $payboxAuthorizationCode;
+
+    /**
+     * @var array|null
+     *
+     * @ORM\Column(type="json_array", nullable=true)
+     */
+    private $payboxPayload;
+
+    /**
+     * @var \DateTimeImmutable|null
+     *
+     * @ORM\Column(type="datetime_immutable", nullable=true)
+     */
+    private $payboxDateTime;
+
+    /**
+     * @var string|null
+     *
+     * @ORM\Column(unique=true, nullable=true)
+     */
+    private $payboxTransactionId;
+
+    /**
+     * @var string|null
+     *
+     * @ORM\Column(nullable=true)
+     */
+    private $payboxSubscriptionId;
+
+    /**
+     * @var \DateTimeImmutable
+     *
+     * @ORM\Column(type="datetime_immutable")
+     */
+    private $createdAt;
+
+    /**
+     * @var Donation
+     *
+     * @ORM\ManyToOne(targetEntity="AppBundle\Entity\Donation", inversedBy="transactions")
+     */
+    private $donation;
+
+    public function __construct(Donation $donation, array $payboxPayload)
+    {
+        $this->donation = $donation;
+        $this->createdAt = new \DateTimeImmutable();
+        $this->payboxPayload = $payboxPayload;
+        $this->payboxResultCode = $payboxPayload['result'];
+
+        if ($transaction = $payboxPayload['transaction'] ?: false) {
+            $this->payboxTransactionId = $transaction;
+        }
+
+        if (isset($payboxPayload['date'], $payboxPayload['time'])) {
+            [
+                'date' => $date,
+                'time' => $time,
+            ] = $payboxPayload;
+            $this->payboxDateTime = \DateTimeImmutable::createFromFormat(
+                'Y/m/d H:i:s',
+                substr($date, 4, 4).'/'.substr($date, 2, 2).'/'.substr($date, 0, 2).' '.$time
+            );
+        }
+
+        if ($authorization = $payboxPayload['authorization'] ?: false) {
+            $this->payboxAuthorizationCode = $authorization;
+        }
+
+        if ($subscription = $payboxPayload['subscription'] ?: false) {
+            $this->payboxSubscriptionId = $subscription;
+        }
+    }
+
+    public function getId(): ?int
+    {
+        return $this->id;
+    }
+
+    public function getPayboxPayloadAsJson(): string
+    {
+        return json_encode($this->payboxPayload, JSON_PRETTY_PRINT);
```

Comment:
Why `JSON_PRETTY_PRINT` ?

Reply:
Don't know,  I just move this method from Donation.

Reply:
why do you move this if it is not used?
can you check and remove unused code pls?